qcow/src/qcow.rs

```diff
@@ -255,10 +255,10 @@ impl QcowHeader {
         let cluster_bits: u32 = DEFAULT_CLUSTER_BITS;
         let cluster_size: u32 = 0x01 << cluster_bits;
         // L2 blocks are always one cluster long. They contain cluster_size/sizeof(u64) addresses.
-        let l2_size: u32 = cluster_size / size_of::<u64>() as u32;
+        let entries_per_cluster: u32 = cluster_size / size_of::<u64>() as u32;
         let num_clusters: u32 = div_round_up_u64(size, u64::from(cluster_size)) as u32;
-        let num_l2_clusters: u32 = div_round_up_u32(num_clusters, l2_size);
-        let l1_clusters: u32 = div_round_up_u32(num_l2_clusters, cluster_size);
```

Comment:
```
marvin:~/src/cloud-hypervisor (main)$ git grep "let l1_clusters = "
qcow/src/qcow.rs:        let l1_clusters = div_round_up_u64(num_l2_clusters, cluster_size);
qcow/src/qcow.rs:            let l1_clusters = div_round_up_u64(u64::from(header.l1_size), cluster_size);
qcow/src/qcow.rs:        let l1_clusters = div_round_up_u64(l2_clusters, cluster_size);
```

Reply:
Oh okay the same calculation is done somewhere else :)

Reply:
@rbradford Thanks，
Yes, there are 9 places that need to be modified， and so I have restored the original representation of l1_clusters.
@sboeuf

Reply:
@danielverkamp Hi, this bit of code is essentially unmodified from crosvm. Do you think this proposed change makes sense? Looks like it can be found empirically with a disk size > 4TiB

Reply:
Thanks, the new math (and better names) look correct to me. The diff is a bit hard to read due to the renames, though, so if it were me, I might split it up into two commits just to make the bugfix part totally clear.

For the original submitter, if you're willing to send this to crosvm too once it's finalized (so the change is correctly attributed), that would be greatly appreciated! (https://crosvm.dev/book/contributing/index.html)